### PR TITLE
docs(features): complete assets page

### PR DESCRIPTION
**Pull request by clemcode**
### 🔗 Linked issue

#4104 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR completes the `guide/features/assets` page of the documentation.

Preview : https://deploy-preview-4563--nuxt3-docs.netlify.app/guide/features/assets

### CODE REVIEW

File: docs/content/2.guide/2.features/3.assets.md [hunk 1]
```diff
@@ -1,10 +1,50 @@
 # Assets
 
-::NeedContribution
+Nuxt uses two directories to handle assets like stylesheets, fonts or images:
+
+- The `assets/` directory contains every asset that you want the build tool (Vite or Webpack) to process.
```
Comment (Atinux) [inline docs/content/2.guide/2.features/3.assets.md:5]: Can we link to the assets directory here?

File: docs/content/2.guide/2.features/3.assets.md [hunk 2]
```diff
@@ -1,10 +1,50 @@
...
+- The `public/` directory content is served at the server root as-is.
```
Comment (Atinux) [inline docs/content/2.guide/2.features/3.assets.md:6]: Can we link to the public dir as well?

File: docs/content/2.guide/2.features/3.assets.md [hunk 3]
```diff
@@ -1,10 +1,50 @@
...
+
+```vue [app.vue]
+<template>
+  <img src="assets/img/nuxt.png" alt="Discover Nuxt 3" />
+</template>
+```
+
+## `public/` directory
+
+The `public/` directory is used as a public server for every asset that build tools don't need to process. This includes files where names can't be changed, for example `robots.txt`, or that you want to be publicly available at a defined URL of your application.
```
Comment (DamianGlowala) [inline docs/content/2.guide/2.features/3.assets.md:30]:
```suggestion
The `public/` directory is used as a public server for static assets that build tools don't need to process. This includes files whose names cannot be changed (for example `robots.txt`), or the ones you want to be publicly available at a defined URL of your application.
```

File: docs/content/2.guide/2.features/3.assets.md [hunk 4]
```diff
@@ -1,10 +1,50 @@
...
+
+You can get a file in the `public/` directory from your application's code or from a browser by the root URL `/`.
+
+::alert{type=info}
+Nuxt can be configured to use a different name for this directory in the [`nuxt.config` file](/api/configuration/nuxt.config#public)
```
Comment (DamianGlowala) [inline docs/content/2.guide/2.features/3.assets.md:35]:
```suggestion
Nuxt can be configured to use a different name for this directory in the [`nuxt.config` file](/api/configuration/nuxt.config#public).
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

File: docs/content/2.guide/2.features/3.assets.md [hunk 5]
```diff
@@ -1,10 +1,50 @@
...
 
-::NeedContribution
+Nuxt uses two directories to handle assets like stylesheets, fonts or images:
+
+- The [`assets/` directory](/guide/directory-structure/assets) contains every asset that you want the build tool (Vite or Webpack) to process.
+- The [`public/` directory](/guide/directory-structure/public) content is served at the server root as-is.
+
+## `assets/` directory
+
+Nuxt uses Vite or Webpack to build and bundle your application. The main function of these build tools is to process JavaScript files, but they can be extended through [plugins](https://vitejs.dev/plugins/) (for Vite) or [loaders](https://webpack.js.org/loaders/) (for Webpack) to process other kind of assets, like stylesheets, fonts or SVG. This step transforms the original file mainly for performance or caching purposes (such as stylesheets minification or browser cache invalidation).
```
Comment (pi0) [inline docs/content/2.guide/2.features/3.assets.md:10]:
```suggestion
Nuxt uses [Vite](https://vitejs.dev/guide/assets.html) or [Webpack](https://webpack.js.org/guides/asset-management/) to build and bundle your application. The main function of these build tools is to process JavaScript files, but they can be extended through [plugins](https://vitejs.dev/plugins/) (for Vite) or [loaders](https://webpack.js.org/loaders/) (for Webpack) to process other kind of assets, like stylesheets, fonts or SVG. This step transforms the original file mainly for performance or caching purposes (such as stylesheets minification or browser cache invalidation).
```

File: docs/content/2.guide/2.features/3.assets.md [hunk 6]
```diff
@@ -1,10 +1,50 @@
...
+
+In your application's code, you can reference a file located in the `assets/` directory by using the `~/assets/` path or `assets/` alias.
```
Comment (pi0) [inline docs/content/2.guide/2.features/3.assets.md:12]: Let's not advocate for ambiguous route (`assets/`)

```suggestion
In your application's code, you can reference a file located in the `assets/` directory by using the `~/assets/` path.
```

File: docs/content/2.guide/2.features/3.assets.md [hunk 7]
```diff
@@ -1,10 +1,50 @@
...
+
+For example, referencing an image file that will be processed if a build tool is configured to handle this file extension:
+
+```vue [app.vue]
+<template>
+  <img src="assets/img/nuxt.png" alt="Discover Nuxt 3" />
+</template>
+```
+
+## `public/` directory
```
Comment (pi0) [inline docs/content/2.guide/2.features/3.assets.md:28]: I would move this section above assets. Seems more easier to understand and then we can continue with assets and why bundling and hashing is sometimes better

File: docs/content/2.guide/2.features/3.assets.md [hunk 8]
```diff
@@ -1,10 +1,50 @@
...
+
+The `public/` directory is used as a public server for static assets that build tools don't need to process. This includes files whose names cannot be changed (for example `robots.txt`), or the ones you want to be publicly available at a defined URL of your application.
```
Comment (pi0) [inline docs/content/2.guide/2.features/3.assets.md:30]: (we can link to directory structure section)

File: docs/content/2.guide/2.features/3.assets.md [hunk 9]
```diff
@@ -1,10 +1,50 @@
 # Assets
 
-::NeedContribution
+Nuxt uses two directories to handle assets like stylesheets, fonts or images:
+
+- The [`assets/` directory](/guide/directory-structure/assets) contains every asset that you want the build tool (Vite or Webpack) to process.
```
Comment (pi0) [inline docs/content/2.guide/2.features/3.assets.md:5]: (the fact is, `assets/` is mostly convention for where to put assets not an auto scanned dir by nuxt. i wish we could little bit make it clear that other directories can be used)

Comment (clemcode) [reply docs/content/2.guide/2.features/3.assets.md:5]: Thinking about this, I think we might switch the two sections, starting with `public/` : 
1. `public/` directory, the static server
2. `assets/` directory, the **convention** for files that you want processed and not served as-is.

what do you think?

File: docs/content/2.guide/2.features/3.assets.md (hunk 7 above)
Comment (clemcode) [reply docs/content/2.guide/2.features/3.assets.md:28]: Great point, so much easier indeed.

**Review by Atinux — APPROVED** [review]
Happy to merge and iterate later on if needed!

**Review by pi0 — APPROVED** [review]

**Pull request merged**